Add test cases for inlining compiler-private items

notriddle opened the pull request:
Closes #106421

This was already fixed by #121218, but now the test cases are added.

Mark-Simulacrum commented:
@bors r+ rollup

bors commented:
:pushpin: Commit b9f0094afa3d99316de6f2fdbe7961c877fab272 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum b9f0094afa3d99316de6f2fdbe7961c877fab272 -->
<!-- homu: {"type":"Approved","sha":"b9f0094afa3d99316de6f2fdbe7961c877fab272","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

matthiaskrgr commented:
@bors r-
needs rebase https://github.com/rust-lang/rust/pull/121565#issuecomment-1962735599

notriddle committed:
Add test cases for inlining compiler-private items

Closes #106421

This was already fixed by
f5d43a052b9eb464e54af819143467954d814a24, but now the test cases are
added.

bors commented:
:pushpin: Commit 24aa34858645e96316bb61f4a095d13486a56892 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 24aa34858645e96316bb61f4a095d13486a56892 -->
<!-- homu: {"type":"Approved","sha":"24aa34858645e96316bb61f4a095d13486a56892","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
